pages/deployments/oidc/oidc_with_aws.md

```diff
@@ -0,0 +1,187 @@
+---
+keywords: oidc, authentication, IAM, roles, AWS
+---
+
+# OIDC with AWS
+
+The [Buildkite Agent's oidc command](/docs/agent/v3/cli-oidc) allows you to request an OpenID Connect (OIDC) token representing the current job. These tokens can be exchanged on federated systems like AWS for an Identity and Access Management (IAM) role with AWS-scoped permissions.
+
+This process uses the following Buildkite plugins to implement OIDC with AWS and Buildkite pipelines:
+
+- [AWS assume-role-with-web-identity](https://github.com/buildkite-plugins/aws-assume-role-with-web-identity-buildkite-plugin)
+- [AWS SSM Buildkite Plugin](https://github.com/buildkite-plugins/aws-ssm-buildkite-plugin)
+
+Learn more about:
+
+- How OIDC tokens are constructed and how to extract and use claims in the [OpenID Connect Core documentation](https://openid.net/specs/openid-connect-core-1_0.html#IDToken).
+
+- Amazon's implementation of OIDC with their federated system in the [AWS OpenID Connect identity provider in IAM documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc.html).
+
+## Step 1: Set up an OIDC provider in your AWS account
+
+First, you'll need to set up an IAM OIDC provider in your AWS account.
+
+Learn more about how to do this in the [Create an OpenID Connect (OIDC) identity provider in IAM](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc.html) page of the AWS documentation.
+
+On this page, as part of the [Creating and managing an OIDC provider (console)](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc.html#manage-oidc-provider-console) process, specify the following values for the:
+
+- **Provider URL**: `https://agent.buildkite.com`
+
+- **Audience**: `sts.amazonaws.com`
+
+## Step 2: Create a new (or update an existing) IAM role to use with your pipelines
+
+Creating new or updating existing IAM roles is conducted through your AWS account.
+
+Learn more about how to do this in the [Creating a role using custom trust policies (console)](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-custom.html) page of the AWS documentation.
+
+As part of this process:
+
+1. Choose the **Custom trust policy** role type.
+
+1. Copy the following example trust policy in the following JSON code block and paste it into a code editor:
+
+    ```json
+    {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Principal": {
+                    "Federated": "arn\:aws\:iam:\:AWS_ACCOUNT_ID\:oidc-provider/agent.buildkite.com"
```

Comment:
Other than `AWS_ACCOUNT_ID`, is anything else in this string value likely to change?

Reply:
Nothing else should change - it's the ARN of the OIDC provider they created in the previous step, and because the provider URL is the one we specify, then I don't think it's possible for it to be named anything different,